Build fails with TraceHeaderWriter warning on GCC 7.1.1

From an empty directory (or at least one without an existing omr directory):
```
git clone https://github.com/eclipse/omr.git && cd omr && ./configure OMRGLUE=$(pwd)/example/glue && make
```

Yields:
```
g++  -I.  -DUT_DIRECT_TRACE_REGISTRATION -I../../include_core -I../../nls -DLINUX -D_REENTRANT -D_FILE_OFFSET_BITS=64 -DJ9HAMMER -c  -MMD -MP -fno-exceptions -fno-rtti -fno-threadsafe-statics -fPIC -ggdb -m64 -Wreturn-type -Werror -Wall -Wno-non-virtual-dtor -O3 -fno-strict-aliasing   -o TraceHeaderWriter.o TraceHeaderWriter.cpp
In file included from FileUtils.hpp:24:0,
                 from TraceHeaderWriter.cpp:25:
In static member function ‘static void Port::omrmem_free(void**)’,
    inlined from ‘RCType TraceHeaderWriter::tpTemplate(FILE*, unsigned int, unsigned int, const char*, const char*, unsigned int, unsigned int, const char*, unsigned int, unsigned int)’ at TraceHeaderWriter.cpp:330:20:
Port.hpp:226:8: error: attempt to free a non-heap object [-Werror=free-nonheap-object]
    free(*ptr);
    ~~~~^~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [../../omrmakefiles/rules.mk:360: TraceHeaderWriter.o] Error 1
make[2]: Leaving directory '/home/lmaisons/jit.source/omr/tools/tracegen'
make[1]: *** [GNUmakefile:259: tools/tracegen] Error 2
make[1]: Leaving directory '/home/lmaisons/jit.source/omr'
make: *** [GNUmakefile:208: tools] Error 2`
```

--- a/tools/tracegen/TraceHeaderWriter.cpp
+++ b/tools/tracegen/TraceHeaderWriter.cpp
@@ -214,8 +214,8 @@ TraceHeaderWriter::tpTemplate(FILE *fd, unsigned int overhead, unsigned int test
 	char *parmString = NULL;
 	char *parmStringNoLeadingComma = NULL;
 	char *pos = NULL;
-	char *testMacro = (char *) "";
-	char *testNop = (char *) "";
+	char *testMacro =  NULL;
+	char *testNop =  NULL;
 	char *testMacroTemplate = (char *)  "#define TrcEnabled_%s  (%s_UtActive[%u] != 0)\n";
 	char *testNopTemplate = (char *) "#define TrcEnabled_%s  (0)\n";
 
@@ -247,6 +247,9 @@ TraceHeaderWriter::tpTemplate(FILE *fd, unsigned int overhead, unsigned int test
 			goto failed;
 		}
 		sprintf(testNop, testNopTemplate, name);
+	} else {
+		testMacro = (char *) "";
+		testNop = (char *) "";
 	}
 
 	for (unsigned int i = 0; i < parmCount; i++) {
@@ -313,7 +316,6 @@ TraceHeaderWriter::tpTemplate(FILE *fd, unsigned int overhead, unsigned int test
 		}
 	}
 
-
 	Port::omrmem_free((void **)&parmString);
 
 	if (test) {
@@ -342,8 +344,8 @@ TraceHeaderWriter::tpAssert(FILE *fd, unsigned int overhead, unsigned int test, 
 	char *parmString = NULL;
 	char *parmStringNoLeadingComma = NULL;
 	char *pos = NULL;
-	char *testmacro = (char *) "";
-	char *testnop = (char *) "";
+	char *testmacro = NULL;
+	char *testnop = NULL;
 	char *testmacrotemplate = (char *)  "#define TrcEnabled_%s  (%s_UtActive[%u] != 0)\n";
 	char *testnoptemplate = (char *) "#define TrcEnabled_%s  (0)\n";
 
@@ -376,6 +378,9 @@ TraceHeaderWriter::tpAssert(FILE *fd, unsigned int overhead, unsigned int test, 
 			goto failed;
 		}
 		sprintf(testnop, testnoptemplate, name);
+	} else {
+		testmacro = (char *) "";
+		testnop = (char *) "";
 	}
 
 	for (unsigned int i = 0; i < parmCount; i++) {
@@ -414,10 +419,21 @@ TraceHeaderWriter::tpAssert(FILE *fd, unsigned int overhead, unsigned int test, 
 		goto failed;
 	}
 
+	if (test) {
+		Port::omrmem_free((void **)&testmacro);
+		Port::omrmem_free((void **)&testnop);
+	}
+
 	Port::omrmem_free((void **)&parmString);
 	return rc;
 
 failed:
+
+	if (test) {
+		Port::omrmem_free((void **)&testmacro);
+		Port::omrmem_free((void **)&testnop);
+	}
+
 	Port::omrmem_free((void **)&parmString);
 
 	return rc;

